### PR TITLE
add get all movies route + test movie page to see how it works

### PR DESCRIPTION
**Pull request by RGBlife**

### CODE REVIEW

File: src/app/utils/api/fetchMovies.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { Movie } from "@prisma/client";
+import { PaginationSchema } from "../paginationFilter";
+
+export const fetchMovies = async ({ limit, page }: PaginationSchema) => {
+  const result = await fetch(`/api/movies/all?limit=${limit}&page=${page}`);
```
Comment (xandert93) [inline src/app/utils/api/fetchMovies.ts:5]: REST API convention for this endpoint is `/api/movies` + query string e.g. 

`/api/movies?limit=${limit}&page=${page}`

Comment (xandert93) [inline src/app/utils/api/fetchMovies.ts:5]: The fetch call's promise resolves with a `response` ✅ not `result` ❌

File: src/app/utils/api/fetchMovies.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+  return (await result.json()) as Movie[];
```
Comment (xandert93) [inline src/app/utils/api/fetchMovies.ts:6]: I find this a little hard to read. I might have done this instead:

`const movies: Movie[] = await res.json()`
`return movies`

Let me know what you think :)

Comment (RGBlife) [reply src/app/utils/api/fetchMovies.ts]: Agreed, I've updated it to your suggestion :)

File: src/app/utils/api/fetchMovies.ts (hunk 1 above)
Comment (RGBlife) [reply src/app/utils/api/fetchMovies.ts]: Updated! :)

Comment (RGBlife) [reply src/app/utils/api/fetchMovies.ts]: Agreed, makes more sense this way.

**Review by xandert93 — APPROVED** [review]
Nice job x

**Pull request merged**